src/LED/led.c

```diff
@@ -0,0 +1,81 @@
+/*
+ * Race Capture Firmware
+ *
+ * Copyright (C) 2016 Autosport Labs
+ *
+ * This file is part of the Race Capture firmware suite
+ *
+ * This is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *
+ * See the GNU General Public License for more details. You should
+ * have received a copy of the GNU General Public License along with
+ * this code. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "array_utils.h"
+#include "mod_string.h"
+#include "led.h"
+#include "led_device.h"
+
+/*
+ * LED names are the same strings as the enums in code.
+ */
+#define LED_NAME_ENTRY(a)	{#a, a}
+static struct led_names {
+        const char *name;
+        const enum led led;
+} led_names[] = {
+        LED_NAME_ENTRY(LED_ERROR),
```

Comment:
That's clever, but it could change the API through an innocent refactoring. What about decoupling it? 

Kind of reminds me the risks of externalizing the numeric enums into our API.